client/components/payment-logo/index.jsx

```diff
@@ -7,20 +7,43 @@
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import React from 'react';
+import { keys } from 'lodash';
+
+const ALT_TEXT = {
+	alipay: 'Alipay',
+	amex: 'American Express',
+	bancontact: 'Bancontact',
+	diners: 'Diners Club',
+	discover: 'Discover',
+	eps: 'eps',
+	giropay: 'Giropay',
+	ideal: 'iDEAL',
+	jcb: 'JCB',
+	mastercard: 'MasterCard',
+	p24: 'Przelewy24',
+	paypal: 'PayPal',
+	placeholder: '',
+	unionpay: 'UnionPay',
+	visa: 'VISA',
+};
+
+const POSSIBLE_TYPES = keys( ALT_TEXT );
 
 class PaymentLogo extends React.Component {
 	static propTypes = {
-		type: PropTypes.string.isRequired,
-		altText: PropTypes.string.isRequired,
+		type: PropTypes.oneOf( POSSIBLE_TYPES ),
+		altText: PropTypes.string,
 		isCompact: PropTypes.bool,
 	};
 
 	render() {
-		const classes = classNames( 'payment-logo', `is-${ this.props.type }`, {
-			'is-compact': this.props.isCompact,
+		const { altText, isCompact, type } = this.props;
+
+		const classes = classNames( 'payment-logo', `is-${ type }`, {
+			'is-compact': isCompact,
 		} );
 
-		return <div className={ classes } aria-label={ this.props.altText } />;
+		return <div className={ classes } aria-label={ altText || ALT_TEXT[ type ] || '' } />;
```

Comment:
We could add property propagation here by inserting `{ ...this.props }` before `className` here. Doing so will allow consumers of this component to pass down whatever properties they'd like to the `div` element.

Reply:
Hi @jsnmoon ,  I'm a little concerned about the use of `{ ...this.props }`. I'd rather specify a finite list of props that can be consumed by PaymentLogo for ease of maintenance. If consumers pass down props that are not documented, would that causes hassles if we want to refactor this component at some time in the future?

Reply:
I think your point is a valid concern. 

Given that the component already handle most use-cases that may involve property propagation (e.g. setting alt text or class name), I think your call to exclude `{ ...this.props }` is the right one to make.